org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java

```diff
@@ -561,8 +565,6 @@ public Void visitClass(ClassTree node, Void p) {
 				}
 			}
 		});
-		// must be 1st thing added to context
-		context.put(DiagnosticListener.class, diagnosticListener);
```

Comment:
The easier fix is to move `TolerantJavaCompiler.configureCompilerInstance(context);` after the L565.